### PR TITLE
Bump macos deployment target; stop holding back `firebase_core`; run `tools/upgrade pub`

### LINKED ISSUE #1116: Can't upgrade firebase_core to 3.4.0, because macOS deployment target
When I try upgrading our dependencies with `tools/upgrade pub`, I get the following error from CocoaPods:

```
+ pod update --project-directory=macos/
Update all pods
Updating local specs repositories
Analyzing dependencies
firebase_core: Using Firebase SDK version '11.0.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '11.0.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/Messaging":
  In Podfile:
    firebase_messaging (from `Flutter/ephemeral/.symlinks/plugins/firebase_messaging/macos`) was resolved to 15.1.0, which depends on
      Firebase/Messaging (~> 11.0.0)

Specs satisfying the `Firebase/Messaging (~> 11.0.0)` dependency were found, but they required a higher minimum deployment target.
```

This appears to correspond to the following release note:
https://firebase.google.com/support/release-notes/ios#version_1100_-_july_30_2024
saying version 11.0.0 of the Firebase Apple SDK started requiring a minimum deployment target of macOS 10.15.

Naturally we're happy to bump up our minimum macOS deployment target — 10.15 is from 2019, and even increasing to last year's macOS 14 would be perfectly fine. But I spent a few minutes just now attempting to do so — editing the `MACOSX_DEPLOYMENT_TARGET = 10.14;` lines in the `project.pbxproj` file so they say 10.15 (or 13.0) instead, then also tried setting it in Xcode which added some new lines of that form — and the error persisted. (Perhaps there's somewhere *else* that `pod update` is looking? Perhaps it has a cache of some kind that it fails to update?)

So further debugging is required.

Meanwhile, we'll be holding back `firebase_core` at 3.3.0 rather than 3.4.0 (and so holding back the other Firebase-related packages too), because that's the upgrade that triggers this.

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes #1116.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Is the CI failure another flavor of #1177?

```
0s
Run flutter precache --universal
fatal: Not a valid object name origin/master
Error: Process completed with exit code [1](https://github.com/zulip/zulip-flutter/actions/runs/12421852640/job/34682388039?pr=1184#step:6:1)28.
```

**Comment by gnprice** [comment]
> Is the CI failure a new flavor of #1177?

Yeah, I think so.

**Comment by gnprice** [comment]
(I'll send a fix for that)

**Comment by chrisbobbe** [comment]
Thanks. I also saw this go by: https://github.com/zulip/zulip-flutter/commit/d90c5a02381a11e93505d11f1cac94d4f6daae61#commitcomment-150583217

**Comment by gnprice** [comment]
Thanks for taking care of this! These changes all look great.

Since we're upgrading `firebase_messaging`, and particularly as the underlying SDK seems to be upgraded by a major version (`FirebaseMessaging (11.0.0)` in the Podfile.lock), we should probably repeat our manual testing that notifications still work. (It'll be great to someday have automated tests for that: #757, #758.) Would you do that?

That testing can be batched with further upgrading Firebase to the latest: if you upgrade `share_plus` and `share_plus_platform_interface`, then all the Firebase packages can be upgraded to latest. No need to manually test on the intermediate upgraded versions, except in the unlikely case that things don't work after fully upgrading.

**Comment by chrisbobbe** [comment]
Mm, good idea. OK to test on Android only, do you think, since on iOS it takes more setup? Then for iOS I can make sure to take a look in the new version from TestFlight when we publish that?

**Comment by chrisbobbe** [comment]
OK, and checked that notifications are still being presented on the office Android device running Android 9.

We'll plan to test iOS notifications when making the next beta release, because Apple makes it complicated to test client-side notification code in dev: https://github.com/zulip/zulip-mobile/blob/main/docs/howto/push-notifications.md#testing-client-side-changes-on-ios

File: test/model/schemas/drift_schema_v2.json [hunk 1]
```diff
@@ -1 +1 @@
...
...e":"email","getter_name":"email","moor_type":"string","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"api_key","getter_name":"apiKey","moor_type":"string","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"zulip_version","getter_name":"zulipVersion","moor_type":"string","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"zulip_merge_base","getter_name":"zulipMergeBase","moor_type":"string","nullable":true,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"zulip_feature_level","getter_name":"zulipFeatureLevel","moor_type":"int","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"acked_push_token","getter_name":"ackedPushToken","moor_type":"string","nullable":true,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]}],"is_virtual":false,"without_rowid":false,"constraints":[],"unique_keys":[["realm_url","user_id"],["realm_url","email"]]}}]}
```
Comment (gnprice) [inline test/model/schemas/drift_schema_v2.json:1]: Huh, a bit awkward that this changes, as its purpose is to be a historical record of how things were when we introduced v2 of the schema.

So I got a diff of what changed:
```
$ diff -U1 \
    <(git show @~:test/model/schemas/drift_schema_v2.json | jq .) \
    <(<test/model/schemas/drift_schema_v2.json jq .)
```

```diff
--- /dev/fd/63	2024-12-20 14:31:03.565983750 -0800
+++ /dev/fd/62	2024-12-20 14:31:03.565983750 -0800
@@ -3,3 +3,3 @@
     "description": "This file contains a serialized version of schema entities for drift.",
-    "version": "1.1.0"
+    "version": "1.2.0"
   },
@@ -24,2 +24,5 @@
             "defaultConstraints": "PRIMARY KEY AUTOINCREMENT",
+            "dialectAwareDefaultConstraints": {
+              "sqlite": "PRIMARY KEY AUTOINCREMENT"
+            },
             "default_dart": null,
```

Seems like no substantive change, so it's fine.

File: test/model/schemas/schema_v1.dart [hunk 2]
```diff
@@ -1,6 +1,6 @@
+// dart format width=80
 // GENERATED CODE, DO NOT EDIT BY HAND.
 // ignore_for_file: type=lint
-//@dart=2.12
 import 'package:drift/drift.dart';
```
Comment (gnprice) [inline test/model/schemas/schema_v1.dart:4]: Similarly, awkward that these `schema_vN.dart` files change, but again the change is a no-op (and this time just `git log -p` is enough to show that).

File: pubspec.yaml [hunk 3]
```diff
@@ -54,8 +54,8 @@ dependencies:
   package_info_plus: ^8.0.0
   path: ^1.8.3
   path_provider: ^2.0.13
-  share_plus: ^9.0.0
-  share_plus_platform_interface: ^4.0.0
+  share_plus: ^10.0.0
+  share_plus_platform_interface: ^5.0.0
```
Comment (gnprice) [inline pubspec.yaml:58]: Done by editing pubspec.yaml and running `tools/upgrade pub` twice.

Huh, why'd you have to run `tools/upgrade` twice? I'd expect it to be idempotent — for the first time to go all the way that a second time would take you. (Modulo new package versions getting uploaded to the package repo in between.)

File: pubspec.lock [hunk 4]
```diff
@@ -363,50 +363,50 @@ packages:
...
     dependency: "direct main"
     description:
       name: firebase_messaging
-      sha256: "1b0a4f9ecbaf9007771bac152afad738ddfacc4b8431a7591c00829480d99553"
+      sha256: "151a3ee68736abf293aab66d1317ade53c88abe1db09c75a0460aebf7767bbdf"
       url: "https://pub.dev"
     source: hosted
-    version: "15.0.4"
+    version: "15.1.6"
   firebase_messaging_platform_interface:
```
Comment (gnprice) [inline pubspec.lock:394]: This should mean that all the Firebase packages are upgraded to latest:
      https://github.com/zulip/zulip-flutter/pull/1184#issuecomment-2555964673

You can confirm this yourself using `flutter pub outdated`. Here's the output I see at the tip of the branch:
```
$ flutter pub outdated
Showing outdated packages.
[*] indicates versions that are not the latest available.

Package Name              Current              Upgradable           Resolvable           Latest   

direct dependencies:     
collection                *1.19.0              *1.19.0              *1.19.0              1.19.1   
device_info_plus          *10.1.2              *10.1.2              11.2.0               11.2.0   
flutter_color_models      *1.4.0 (overridden)  *1.4.0 (overridden)  *1.4.0 (overridden)  1.3.3+2  
intl                      *0.19.0              *0.19.0              *0.19.0              0.20.1   
mime                      *1.0.6               *1.0.6               2.0.0                2.0.0    
wakelock_plus             *1.2.8               1.2.9                1.2.9                1.2.9    

dev_dependencies:        
flutter_lints             *4.0.0               *4.0.0               5.0.0                5.0.0    
json_serializable         *6.9.0               *6.9.0               *6.9.0               6.9.2    
pigeon                    *20.0.2              *20.0.2              22.7.0               22.7.0   
stack_trace               *1.12.0              *1.12.0              *1.12.0              1.12.1   

transitive dependencies: 
characters                *1.3.0               *1.3.0               *1.3.0               1.4.0    
matcher                   *0.12.16+1           *0.12.16+1           *0.12.16+1           0.12.17  
material_color_utilities  *0.11.1              *0.11.1              *0.11.1              0.12.0   
meta                      *1.15.0              *1.15.0              *1.15.0              1.16.0   
stream_channel            *2.1.2               *2.1.2               *2.1.2               2.1.3    
string_scanner            *1.4.0               *1.4.0               *1.4.0               1.4.1    
term_glyph                *1.2.1               *1.2.1               *1.2.1               1.2.2    
vm_service                *14.3.1              *14.3.1              *14.3.1              15.0.0   
win32_registry            *1.1.5               *1.1.5               *1.1.5               2.0.1    

transitive dev_dependencies:
_fe_analyzer_shared       *76.0.0              *76.0.0              *76.0.0              78.0.0   
analyzer                  *6.11.0              *6.11.0              *6.11.0              7.1.0    
analyzer_plugin           *0.11.3              *0.11.3              *0.11.3              0.12.0   
dart_style                *2.3.7               *2.3.7               *2.3.7               3.0.1    
lints                     *4.0.0               *4.0.0               5.1.1                5.1.1    
pubspec_parse             *1.3.0               1.4.0                1.4.0                1.4.0    
source_gen                *1.5.0               *1.5.0               *1.5.0               2.0.0    
yaml                      *3.1.2               3.1.3                3.1.3                3.1.3    

3 upgradable dependencies are locked (in pubspec.lock) to older versions.
To update these dependencies, use `flutter pub upgrade`.

5  dependencies are constrained to versions that are older than a resolvable version.
To update these dependencies, edit pubspec.yaml, or run `flutter pub upgrade --major-versions`.
```

I think most of those are probably from the Flutter version being a couple of weeks old. In any case, no Firebase packages in the list.

In the terminal it comes in helpful color, too. Here's a screenshot:
image

**Review by gnprice — COMMENTED** [review]
Thanks! The changes all look good. Some comments/questions about the commit message on the last commit.

File: pubspec.yaml (hunk 3 above)
Comment (PIG208) [reply pubspec.yaml:58]: (Linking the follow-up to this on CZO [here](https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/deps.3A.20running.20.60tools.2Fupgrade.20pub.60.20twice))

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by PIG208** [comment]
Looks like we cloned the right version of Flutter in CI, but 38575 from 3.28.0-2.0.pre.38575 is way too large for the version to be inferred correctly as we set `--depth=1000`:
```
  git clone --depth=1000 -b main https://github.com/flutter/flutter ~/flutter
```

**Comment by chrisbobbe** [comment]
I'm not sure offhand what would cause such a large number there.

**Comment by gnprice** [comment]
The large number is because of the merge of flutter/engine into the flutter/flutter monorepo. It's counting all the migrated engine commits, because they're part of the history of the new commit but not the history of that tag. Here's `git log --oneline --graph` there:
```
* 62c6859e5 (origin/master, origin/main, origin/HEAD) Delete unused `engine_hash.dart` script (and test), simplify `engine_hash.sh`. (#160549)
* 65ff06028 Remove bringup: true from web tests. (#160788)
* 5c7d9d01b [Impeller] fix device buffer nullptr on Intel macOS. (#160645)
[…]
* b79ee9282 [tech debt] remove web engine windows build (#160573)
* 0fcb6aa5b Upstream and Origin check (#160574)
* d14259a25 Commit engine.realm as empty (#160541)
* f0bf49559 Trigger Build Part Deux (#160481)
* 1398dc7ee Trigger Build (#160476)
*   721b47e7e Merge flutter/engine into framework
|\  
| * 2a17b7544 Roll Dart SDK from a74650e32880 to 37bbc285d8d2 (1 revision) (flutter/engine#57244)
| * 7f8da94c2 embedder: fix bit-order in software pixel format description (flutter/engine#57156)
| * fe7d6f391 [monorepo] remove Linux Web Framework tests (flutter/engine#57234)
| * 7db982f38 Roll Skia from dca47129cf2e to f784f78a5669 (3 revisions) (flutter/engine#57242)
| * d2a5b9dcb [web] Reland: (Add `crossOrigin` property to `<img>` tag used for decoding)++ (flutter/engine#57228)
| * 37a7c44c1 Roll Skia from 804042d75299 to dca47129cf2e (1 revision) (flutter/engine#57241)
[…]
```

… and that branch merged in from the right is the engine commits and goes on for over 30000 of them.

Or as `gitk` shows it:
image

**Comment by gnprice** [comment]
I think the fix is that before doing that Flutter upgrade, a prep commit should drop the `--depth=1000`. It'll make CI a little slower; but we can revert after the next Flutter upgrade if there's a new tag by then.

**Comment by chrisbobbe** [comment]
Makes sense, thanks for explaining! Revision pushed, with that new commit.

**Comment by gnprice** [comment]
Thanks! All now LGTM except I think you did a `git commit` when you wanted `git commit --amend`:
c5cbf5856 ci: Drop --depth=1000 in cloning Flutter
2497dfb82 ci: Remove --depth=1000 in doing Flutter clone, to be added back later

I'll squash those and merge.

**Pull request merged**